### PR TITLE
feat: improve scrolling behavior and layout consistency with banner

### PR DESCRIPTION
**Pull request by karrui**
## Solution

This PR addresses layout issues related to scrolling and height calculations across multiple components in the studio app when the banner is visible.

**Breaking Changes**

- [x] No - this PR is backwards compatible

**Improvements**:

- Updated height calculations in various components to use `height={0}` and `minH="100%"` instead of fixed `100vh` values
- Improved scrolling behavior in modal screens, sidebars, and page layouts
- Enhanced responsiveness of layout components

**Bug Fixes**:

- Fixed potential overflow issues in CmsSidebar components
- Resolved layout inconsistencies in CreatePageModal screens
- Corrected height calculations in various page templates

## Before & After Screenshots
Should be no different in snapshots, but scrolling behaviour should now be consistent.

## Tests

- Test scrolling behavior in CmsSidebar components
- Verify layout consistency in CreatePageModal screens
- Check responsiveness of page templates and layouts
- Ensure proper rendering of content in various viewport sizes

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/733?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#733** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/733?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#729** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/729?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
I'll fix the stories; failing because snapshot not actually expanding the heights hehe

**Comment by sehyunidaaa** [comment]
@karrui one unreviewed story has a clarifying comment! feel free to accept that story if my assumption is wrong. otherwise lgtm 🙏

File: apps/studio/src/features/editing-experience/components/CreatePageModal/DetailsScreen.tsx [hunk 1]
```diff
@@ -102,84 +102,86 @@ export const CreatePageDetailsScreen = () => {
...
-                />
-                {errors.title?.message ? (
-                  <FormErrorMessage>{errors.title.message}</FormErrorMessage>
-                ) : (
-                  <FormHelperText mt="0.5rem" color="base.content.medium">
-                    {MAX_TITLE_LENGTH - title.length} characters left
-                  </FormHelperText>
-                )}
-              </FormControl>
+          <Stack height={0} minH="100%" overflow="auto">
```
Comment (seaerchin) [inline apps/studio/src/features/editing-experience/components/CreatePageModal/DetailsScreen.tsx:105]: `height = 0 + minH=100%` to force the height to always be 100%

**Review by seaerchin — APPROVED** [review]
lgtm! i think the `height` + `minH` works becuase of `flex` - see [this article](https://www.joshwcomeau.com/css/interactive-guide-to-flexbox/)

**Pull request merged**